### DISCUSSION TITLE
SPA example has trouble with auto-generated '_nicegui' URLs

### DISCUSSION THREAD

**Discussion by alexzaech** [zauberzeug/nicegui#2035] [Q&A]
### Description

Using the default SPA example from https://github.com/zauberzeug/nicegui/tree/main/examples/single_page_app/ and adding an `ui.image()` loading an image source from the local file system does not work. 

Example:
```python
...
@router.add('/')
    def show_one():
        ui.label('Content One').classes('text-2xl')
        ui.image("./dummy.jpg").classes("w-64"). # Testing...
...
```

Modifying the code a bit further to dump the path parameter

```python
...
@ui.page('/{x:path}')
def main(x:str|None = None):
    router = Router()
    print("x")
...
```
prints out `_nicegui/auto/static/...`. 

The routing of `_nicegui/auto/static/...` using a path parameter containing `{_:path}` to match _any path_ along '`/`' as required for SPA seems to cause the issue here. Maybe this is somehow related to #1854

Loading the image source from a different route as with `app.add_static_files("/files", ".")` or running the entire SPA from a route different from '`/`' (e.g., '`/home`') doesn't show this issue.

**Comment by falkoschindler** [comment]
Oh I see, `@ui.page('/{_:path}')` matches our auto-generated image path `_nicegui/auto/static/...`.

I wonder how we can avoid this conflict elegantly. Serving files via `add_static_files`, as you suggested, show definitely work.

Alternatively you can use
```py
@ui.page('/')
@ui.page('/two')
@ui.page('/three')
```
instead of a catch-all route.

Since it is only a simple example showing the principle of a basic router, I don't think there's anything broken or missing. If there's a simple way to avoid this issue, we can fix it. Otherwise I'd leave the example as it is.

**Comment by alexzaech** [comment]
Many thanks for your feedback @falkoschindler!
I'm new to NiceGui, and this just raised while playing around. There is nothing wrong with the example itself. A hint in the example would be helpful. But this GitHub issue might be valuable enough for others and serves as documentation.

Anyway, really cool library 👍
